lib/src/main/java/io/ably/lib/types/BaseMessage.java

```diff
@@ -78,7 +78,12 @@ public void decode(ChannelOptions opts) throws AblyException {
 					if(!match.matches()) break;
 					String xform = match.group(1).intern();
 					if(xform == "base64") {
-						data = Base64Coder.decode((String)data);
+						try {
+							data = Base64Coder.decode((String) data);
+						}
+						catch (IllegalArgumentException e) {
+							throw AblyException.fromThrowable(e);
```

Comment:
Without rethrowing the lib would crash on IllegalArgumentException. Exceptions is caught in Channel.onMessage or Channel.onPresense and logged. Rethrowing is used in the same method for JSON case.